### ISSUE TITLE
Install won't complete

### ISSUE THREAD

**Issue by coleherzer** [felixrieseberg/windows-build-tools#121]
I've been trying to run the install ever since yesterday morning and am still running into issues with the completion of the installation.

Note: I've run the command multiple times and have at times Control+C'd out of the process (after 20-30+ mins of no progress on the install) and then closed out the window and have restarted my computer to try to run the command again. I've also run npm uninstall windows-build-tools a few times here and there before/after an attempt to see if that would help (it didn't). 

I've tried a bunch of solutions including the below solutions I've see on other threads:
1. Tried removing the .windows-build-tools file in and then using `npm --add-python-to-path='true' --debug install --global windows-build-tools` to run the install. This didn't seem to do anything particularly helpful. 
2. I've also most recently tried to run the install and then in another admin cmd window cd into %USERPROFILE%\.windows-build-tools and then running `msiexec /n /lvx+! python-log.txt /i python-2.7.14.amd64.msi /quiet` which seems to allow the python-log.txt file to grow, but that was 20+ mins ago and the install is still running at the exact same place. 
3. I've also tried uninstalling python and re-running the command and that doesn't seem to help either
4. Tried Manually installing Visual Studio Build Tools as well and then re-tried the command

Anything in particular I may be missing or should try??

OS: Windows 10 Pro
Node Version: 10.3.0

Latest status from installers (hasn't changed for 30+ mins):
Status from the installers:
  windows-build-tools Tail: Waiting for log file to appear in C:\Users\cherzer\.windows-build-tools\build-tools-log.txt +0ms
  windows-build-tools Tail: Waiting for log file to appear in C:\Users\cherzer\.windows-build-tools\python-log.txt +1ms
  windows-build-tools Tail: waitForFile: still waiting +1ms
  windows-build-tools Tail: waitForFile: still waiting +0ms

**Comment by coleherzer** [comment]
Just a note if there's anyone else running into issues in the future: 

Node v10.3.0 doesn't seem to work well with the installer. I used nvm-windows to switch my node version to a later version (I used 8.11.0) and the install went through and finished with 0 issues after only a minute or two.

**Comment by fabiomor** [comment]
hi,
did it occurred to you to be stuck during visual studio tool installation without any error/warning?

tnx

**Comment by Landen13** [comment]
same problem +1

**Comment by jscissr** [comment]
I have the same problem on Windows 7.
With Node.js 10.3.0 the installation is stuck with this output:
```
---------- Visual Studio Build Tools ----------
---------- Visual Studio Build Tools ----------
---------- Visual Studio Build Tools ----------
---------- Visual Studio Build Tools ----------
[1B18:1D54][2018-06-06T16:05:14]i371: Updating session, registration key: SOFTWARE\Microsoft\Windows\CurrentVe
rsion\Uninstall\{a9528995-e130-4501-ae19-bbfaddb779cc}, resume: ARP, restart initiated: No, disable resume: No

[1B64:1B50][2018-06-06T16:05:14]i399: Apply complete, result: 0x0, restart: None, ba requested restart:  No

------------------- Python --------------------
Successfully installed Python 2.7
```

With Node.js 8.11.2 the install worked.

**Comment by bferdinandus** [comment]
this looks like a duplicate from #116

**Comment by coleherzer** [comment]
Yes @bferdinandus it's a very similar issue. I didn't see #116 until now and also didn't realize my issue was with the node version I was using until after I had made the post. 

@fabiomor @Landen13 Yeah, that's more or less what I was running in to. Try switching your version of node back to an early 8 version and see if that works. If not, I'm probably the wrong person to ask as I barely figured it out myself lol. Good luck

**Comment by fabiomor** [comment]
tnx guys

**Comment by ilpanda** [comment]
I   solved this problem  。 #116 

Then I started the taskmanager to kill the "BuildTools_Full.exe" (Don't close your  cmd)
Then I added the following to the "build-tools-log.txt" : Variable: IsInstalled = 1

then the npm install of windows build tools finished with success.

Hope this helps.

**Comment by felixrieseberg** [comment]
Thank you all so much for the reporting, this helps a lot. The issue should now be entirely fixed (or, at least I hope so, I had to apply some workarounds to bend Visual Studio to our collective will).

**Comment by NoobCoding94** [comment]
> I solved this problem 。 #116
> 
> Then I started the taskmanager to kill the "BuildTools_Full.exe" (Don't close your cmd)
> Then I added the following to the "build-tools-log.txt" : Variable: IsInstalled = 1
> 
> then the npm install of windows build tools finished with success.
> 
> Hope this helps.

I tried this method thanks to this comments and it's works! Thanks.

**Comment by shahkeval0101** [comment]
unable to find BuildTools_Full.exe" this at task manager processes..
